3.4.x.g2

gerhardol opened the pull request:
- Includes the changes in 3.4.x.g, but reimplemented
- Android Studio support. (see also #18) ADT is still supported, only aidl files moved and AndroidManifest.xml duplicated. When ADT support is dropped, move directories and update app/build.gradle
- Unconfigured keys are not handled at all. Note that there is no check if there actually is any mapping for the key, that would require a config change to support combo.
- Keep original KeyEvent for default dispatching
- Allow external devices by default - some devices identify normal buttons as external
- SeekBarPreference corrections

shayneholmes committed:
Allow context: apps running over lockscreen

Previously, any app running over the lockscreen just has the context
"guard". With this change, the lockscreen itself reports as "guard", but
apps (like Camera) will use their own contexts.

Commit:
Merge pull request #13 from shayneholmes/context-switch-on-apps-over-lockscreen

Allow context: apps running over lockscreen (lock screen camera)

gerhardol committed:
Merge branch '3.4.x' of https://github.com/SpazeDog/xposed-additions into 3.4.x

Conflicts:
	src/com/spazedog/xposed/additionsgb/backend/pwm/PhoneWindowManager.java

gerhardol committed:
Merge branch '3.4.x' of https://github.com/SpazeDog/xposed-additions into 3.4.x

gerhardol committed:
Suppress lint warning (since ADT 21)
(uses the default anyway, allow app backup using adb - not a security concern here)
Very minor change

gerhardol committed:
Add .gitignore, to avoid seeing generated files when committing

gerhardol committed:
Add .classpath, to simplify setup (especially when reapplying changes to new branches)

gerhardol committed:
* Power long press:
https://github.com/SpazeDog/xposed-additions/commit/fb6c8ddd6f098b36491cc6ca68c4108e6b44718b
I am OK with the implementation, but would prefer a slightly different implementation (done in my implementation):
 * Abort the handling at any other key event, not just power presses
 * Get the vibrate feedback a little after the normal long-long press timeout, not just before reset
 * Increase time from 8s (well 8s+2.5s include vibrate) to 15s to always be longer than the default 8-12s.

from 3.3.x.g: https://github.com/gerhardol/xposed-additions/commit/43876c12d4a1908fb58b1c5a8fd4fe4cda76419d

gerhardol committed:
debug printouts in pwm hooks:
Add a short format time stamp (<10s), to simplify timeout debugging
Include the invoked event for long-press/click actions (long-press printed default or not, click used two lines)

gerhardol committed:
Use policyFlagsPos to get policyFlags

gerhardol committed:
Minor spelling in debug printout

gerhardol committed:
Condition check for non-Pro was incorrect
Also simplification of some of ?: constructs, that are hard to follow.

gerhardol committed:
Non essential merge, no functionality change: Use one mKeyActions instead of mClickActions, mPressActions
This slightly simplifies the code, for instance handles the non-Pro overrides in one place.

gerhardol committed:
Repeating (default long-press) should occur immediately after the time out, so insert first repeat immediately. The first timeout should be shorter than hook_viewConfigTimeouts too.
repeatCount() is not needed in the check

gerhardol committed:
No functionality change, wrapping some calls, actually using mRepeatCount in preparation for long-long press

gerhardol committed:
Allow combi events only using last (secondary) key (as is now) or fully repeated
So event mapped to Power-Home double click can be invoked
primary-down,sec-down, up*2, prim-down, sec-down, up*2
  is same as
primary-down,sec-down, sec up, sec-down

Similar as the following for 3.3: https://github.com/gerhardol/xposed-additions/commit/322cfa667c838f9fe7f21cd115727bef927d890b

gerhardol committed:
http://robobunny.com/wp/2013/08/24/android-seekbar-preference-v2/
Comment 12, 15

to avoid printouts like

W/View    ( 4916): requestLayout() improperly called by android.widget.LinearLayout{52861f50 V.E..... ......ID 11,18-166,41 #7f0b0009 app:id/seekBarPrefBarContainer} during layout: running second layout pass

gerhardol committed:
For tap/long-press timeouts: Set max to 50 ms over current val if that is lower than hardcoded (set in xml).
This allows increasing the max value. More than 50 can be set by exit/enter several times. (inconvenient, but most users seem to want lower val.)
(This could be done by just adding 25% or so to current)
Similar could be done for min.

For some devices, 200ms for the double click is too little.

gerhardol committed:
Merge branch '3.4.x.timeout_remap' into 3.4.x.g

* 3.4.x.timeout_remap:
  For tap/long-press timeouts: Set max to 50 ms over current val if that is lower than hardcoded (set in xml). This allows increasing the max value. More than 50 can be set by exit/enter several times. (inconvenient, but most users seem to want lower val.) (This could be done by just adding 25% or so to current) Similar could be done for min.
  http://robobunny.com/wp/2013/08/24/android-seekbar-preference-v2/ Comment 12, 15

gerhardol committed:
Long-long press: For buttons mapped to long-press, create long-press actions if continue to press further 2* the first press time.

Non-trivial merge. Some restructure was needed, even if I tried to keep it down.
(Some commented out functions in EventKey)
the key insertion in handleKeyAction() should probably be lifted to pwm

Note about two TODO in registerKey() and hook_interceptKeyBeforeQueueing()
(I commented out some code that should not be needed)

gerhardol committed:
Merge branch '3.4.x' of https://github.com/SpazeDog/xposed-additions into 3.4.x

gerhardol committed:
Merge branch '3.4.x' into 3.4.x.g

Using 5a080d2 instead of 6ef0a41 for non-pro mapping

gerhardol committed:
Merge branch '3.4.x' into 3.4.x.timeout_remap

gerhardol committed:
Revert "For tap/long-press timeouts: Set max to 50 ms over current val if that is lower than hardcoded (set in xml)."

This reverts commit 8858d163d8b473e9acfe6567f21c5bb19f3578d9.

gerhardol committed:
Pretty printing additions in  	832c5ee

gerhardol committed:
Increase max for timeouts, hardcoded:
delay_key_tap_preference: 200->400
delay_key_press_preference: 500->600

gerhardol committed:
Merge branch '3.4.x.timeout_remap' into 3.4.x.g

gerhardol committed:
Private files for private build

gerhardol committed:
Extend Pro license to 2014-10-17

gerhardol committed:
Increase license to 2014-12-31

gerhardol committed:
Incorrect resource files.
These changes are supposedly not required in ADT but gives cryptic failures in Android Studio.

To convert to AS:
* Import the ADT project in AS, this copies the project to a new location
* Remove app/assets/xposed_init
* Copy the file XposedBridgeApi-36.apk (in old project) to new location: app/libs/XposedBridgeApi.apk
* Add to app/build.gradle (below reflecttools):     compile files('libs/XposedBridgeApi.jar')
* Add to .gitignore as below (all of .idea can be added instead of the individual directories)

.gradle
/local.properties
.DS_Store
/build

#IntelliJ private
/*.iml

# User-specific stuff:
.idea/workspace.xml
.idea/tasks.xml
.idea/dictionaries

# Sensitive or high-churn files:
.idea/dataSources.ids
.idea/dataSources.xml
.idea/sqlDataSources.xml
.idea/dynamic.xml
.idea/uiDesigner.xml

# Gradle:
.idea/gradle.xml
.idea/libraries

gerhardol committed:
Merge branch '3.4.x.as' into 3.4.x.g.p

gerhardol committed:
Files added by migration to Android Studio
The structure is reverted to ADT. However, AndroidManifest.xml had to be copied to app directory
Note that the bundled support-v4 dir was removed
Also readded .gitignore

gerhardol committed:
Merge branch '3.4.x.as.4' into 3.4.x.g.p

gerhardol committed:
Move files to work in AS

gerhardol committed:
Merge branch '3.4.x.as.4' into 3.4.x.g.p.2

gerhardol committed:
Initial commit to skip unconfigured primary keys
Partly working

gerhardol committed:
Update to AndroidStudio 1.0rc2
Previous AS build did not run

gerhardol committed:
Merge branch '3.4.x.as.4' into 3.4.x.g.p.2

gerhardol committed:
Permanently unlock Pro license, but add "buy Pro banner" unless installed

gerhardol committed:
Set remap externals as default
Change skip_unconfigured_primary_key to change_unconfigured_primary_key (so default is reversed)

gerhardol committed:
Bump for release

gerhardol committed:
Change printout for temporary Pro extension

gerhardol committed:
Non-Pro could not have other than key actions
For CHECK_UNCONFIGURED_PRIMARY_KEY: Only wait until first keyUp
Prepare for automatic handling of CHECK_UNCONFIGURED_PRIMARY_KEY

gerhardol committed:
Automatic multidelay: Check if keys are configured before delaying

gerhardol committed:
Remove configuration for CHECK_UNCONFIGURED_PRIMARY_KEY, no longer needed

gerhardol committed:
Android Studio 1.0 - gradle plugin update

gerhardol committed:
Update private version

gerhardol committed:
Major change for the statehandling. Intermediate commit, not fully working. (Will limit the implementation some, to get something maintainable.)

Longer time for combo: Previously the all had to be perform before first longpress timeout

Some better detecting of non existing combo -> default, new/default
Use KeyEvent, avoid passing many individual states
KeyEvent used as session key, to detect done
Remove cache handling of key, not needed
Readded REPEATING state again (more similar to original code)
removed extra time printouts
Attempt to support GingerBread still

gerhardol committed:
Code Analysis warning: possible exception

gerhardol committed:
Major changes
Rewrote the state handling to handle combo insert/delete better

gerhardol committed:
Invoke dispatched keys also in PhoneWindowManager - needed to synch states
Some cleanups
No synchronized() needed in EventManager.
Regression: Invoked-Repeating up could be sent twice

gerhardol committed:
Bump for release

gerhardol committed:
Bump for release

gerhardol committed:
Revert some whitespace differences compared to the original sources

gerhardol committed:
Fix for GingerBread

gerhardol committed:
Cleanup for repeating and injected keys

gerhardol committed:
Do not aort when releasing primary key in a combo

gerhardol committed:
Wake also when screen off also for hw wake keys
Refactoring to handle Call button
Ignore unconfigured keys, also if they have been previously configured

gerhardol committed:
Comment for ACTION_MULTIPLE

gerhardol committed:
Bump for release

gerhardol committed:
Temporarily revert changes to prepare for merge

gerhardol committed:
Reapply private changes

gerhardol committed:
Return but not DISPATCH_ALLOW for allowed actions, to process further

gerhardol committed:
Merge branch '3.4.x.g2' into 3.4.x.g2.p

gerhardol committed:
Bump for release

gerhardol committed:
Further dispatch allow changes

gerhardol committed:
Support for app shortcuts

gerhardol committed:
Bump for release

gerhardol committed:
Compile error

gerhardol committed:
Merge branch '3.4.x.g2.p' into 3.4.x.g2, manually excluding private changes
App shortcuts etc

gerhardol commented:
Added app shortcuts in latest update

dbergloev commented:
Well I am done with exams for now, so I will have a look at it. I also have a few Android 5 changes that needs to be done, before XposedBridge is ported to it. And maybe a new UI not based on the settings interface.

gerhardol committed:
Callcode configuration was not overriding default action
Do not turnon screen for (especially default) keys, can cause wakeup problems
Always return, not DISPATCHING_ALLOW (to get other handling)

gerhardol committed:
* Increase priority slightly for button queue/dispatch, to decrease risk for conflicts. GravityBox uses max for dispatch and default for queueing, so at least not the same.
 * Do only setResult for non PENDING queueing, to keep default in more situations

gerhardol committed:
longpress feedback should have keyEvent as an argument

gerhardol committed:
Merge branch '3.4.x.g2.p' into 3.4.x.g2

gerhardol committed:
Revert private change

gerhardol committed:
Bump for release

gerhardol committed:
Allow Camera, Back remap also when screen off. May be remapped in apps, even if keys make no sense by themselves

gerhardol committed:
Invoked keys: Do not delay first down until up or long-long timeout. No big difference as (most) action first when up/timeout anyway.

gerhardol committed:
Only run default action for secondary in a combo. Order cannot be maintained anyway when aborted. Default Long-press will give longpress (alternating) for both, unexpected. Kepp default click the same, ignoring primary. The code was quite messy too.

gerhardol committed:
Bump for release

gerhardol committed:
Merge branch '3.4.x.g2.p' into 3.4.x.g2

gerhardol committed:
Power key vibrate was to close to normal haptic feedback

gerhardol committed:
No need to wait for changes if no combo
This should be a common case, for instance if no actions in screen off

gerhardol committed:
For unconfigured single keys, check if the conditions for thr combo matches. This is similar to the previous commit, decreases wait especially when screen is off.

gerhardol committed:
Merge branch '3.4.x.g2.p' into 3.4.x.g2

gerhardol committed:
Bump for release

gerhardol committed:
Merge branch '3.4.x' into 3.4.x.g2.p

gerhardol committed:
Android Studio related changes, in addition to commit 707aa96

gerhardol committed:
707aa96 did not handle API 17 (4.2) correctly for goToSleep so there was an exception at init
Note: the method is not used by the module

gerhardol committed:
Merge branch '3.4.x.g2.p' into 3.4.x.g2

gerhardol committed:
Key mapping with conditions: Use default (screen) condition if no action set

gerhardol committed:
For application conditions, add configuration for 'Use screen on condition for unconfigured actions'

gerhardol committed:
Set "use default condition" to false by default

gerhardol committed:
Use GUI config for "use default condition" in code

gerhardol committed:
Merge branch '3.4.x.g2.p' into 3.4.x.g2

gerhardol commented:
Too late to merge this now, I will not support a merge
